py-polars/src/conversion.rs

```diff
@@ -599,7 +599,24 @@ fn materialize_list(ob: &PyAny) -> PyResult<Wrap<AnyValue>> {
 
 impl<'s> FromPyObject<'s> for Wrap<AnyValue<'s>> {
     fn extract(ob: &'s PyAny) -> PyResult<Self> {
-        let type_name = ob.get_type().name()?;
+        let mut type_name = ob.get_type().name()?;
+        // Can't use pyo3::types::PyDateTime with abi3-py37 feature,
+        // so need this workaround instead of `isinstance(ob, datetime).
+        if (type_name != "datetime") && (type_name != "date") {
```

Comment:
Can we do this more expensive check later in the function? I want have fast conversion for the primitive types like int, float, bool, str etc.